packages/buidler-core/src/internal/cli/analytics.ts

```diff
@@ -0,0 +1,268 @@
+      // https://support.google.com/tagmanager/answer/6164990
+      //
+      // Custom dimension 1: Project Id
+      // This is computed as the keccak256 hash of the project's absolute path.
+      cd1: this._projectId,
+      // Custom dimension 2: User type
+      //   Possible values: "CI", "Developer".
+      cd2: this._userType,
+      // Custom dimension 3: Buidler Version
+      //   Example: "Buidler 1.0.0-beta.8".
+      cd3: await getBuidlerVersion()
+    };
+  }
+
+  private _sendHit(hit: RawAnalytics): [AbortAnalytics, Promise<void>] {
+    log(`Sending hit for ${hit.dp}`);
+
+    const controller = new AbortController();
+
+    const abortAnalytics = () => {
+      log(`Aborting hit for ${JSON.stringify(hit.dp)}`);
+
+      controller.abort();
+    };
+
+    const hitPayload = qs.stringify(hit);
+
+    log(`Hit payload: ${JSON.stringify(hit)}`);
+
+    const hitPromise = fetch(googleAnalyticsUrl, {
+      body: hitPayload,
+      method: "POST",
+      signal: controller.signal
+    })
+      .then(() => {
+        log(`Hit for ${JSON.stringify(hit.dp)} sent successfully`);
+      })
+      // We're not really interested in handling failed analytics requests
+      .catch(() => {
+        log("Hit request failed");
+      });
+
+    return [abortAnalytics, hitPromise];
+  }
+
+  /**
+   * Checks whether we're using Buidler in development mode (that is, we're working _on_ Buidler).
+   * We don't want the tasks we run at these moments to be tracked, so we disable analytics if so.
+   */
+  private _isLocalDev(): boolean {
+    const executionMode = getExecutionMode();
+
+    return (
+      executionMode === ExecutionMode.EXECUTION_MODE_LINKED ||
+      executionMode === ExecutionMode.EXECUTION_MODE_TS_NODE_TESTS
+    );
+  }
+}
+
+// TODO: Check Windows support for this approach
+const globalBuidlerConfigFile = path.join(
+  os.homedir(),
+  ".buidler",
+  "config.json"
+);
+
+async function getClientId() {
+  await fs.ensureFile(globalBuidlerConfigFile);
+
+  let clientId;
+
+  log(`Looking up Client Id at ${globalBuidlerConfigFile}`);
+  try {
+    const data = JSON.parse(await fs.readFile(globalBuidlerConfigFile, "utf8"));
+
+    clientId = data.analytics.clientId;
+
+    log(`Client Id found: ${clientId}`);
+  } catch (e) {
+    log("Client Id not found, generating a new one");
+    clientId = uuid();
+
+    await fs.writeFile(
+      globalBuidlerConfigFile,
+      JSON.stringify({
+        analytics: {
+          clientId
+        }
+      }),
+      "utf-8"
+    );
+
+    log(`Successfully generated clientId ${clientId}`);
+  }
+
+  return clientId;
+}
+
+function getProjectId(rootPath: string) {
+  log(`Computing Project Id for ${rootPath}`);
+
+  const projectId = keccak256(rootPath).toString("hex");
+
+  log(`Project Id set to ${projectId}`);
+  return projectId;
+}
+
+function getUserType(): string {
+  return ci.isCI ? "CI" : "Developer";
+}
+
+function getOperatingSystem(): string {
```

Comment:
This doesn't track the OS version, right? Can we add a comment about this?

Reply:
That's right, it doesn't. I couldn't find a way to reliably do this in Node (the versions reported by the various `os.platform()`, `os.release()`, etc) are not what GA expects, so I took the compromise of just reporting OSX/Linux/Windows for now. Let me know if you know a better way

Reply:
I supposed that was the reason. I'm ok with that. Let's just add a comment explaining it.